File: leetcode/javascript/002418.sort-the-people.js
/**
 * @param {string[]} names
 * @param {number[]} heights
 * @return {string[]}
 * Time complexity: O(n(log(n)))
 * Space complexity: O(n)
 */
var sortPeople = function (names, heights) {
  return names.map((name, i) => [name, heights[i]])
    .sort((a, b) => b[1] - a[1]) // descending
    .map(([name]) => name);
};

File: leetcode/python/002418.sort-the-people.py
class Solution:
    # Time complexity: O(n(log(n)))
    # Space complexity: O(n)
    def sortPeople(self, names: list[str], heights: list[int]) -> list[str]:
        return [
            el[0]
            for el in sorted(zip(names, heights), key=lambda x: x[1], reverse=True)
        ]
